apps/extension/src/popup.tsx

```diff
@@ -23,38 +23,53 @@ function Popup() {
         " text-sm font-sans"
       }>
       <NavProvider>
-        <ModelProvider>
+        <ConfigProvider>
           <NavFrame />
-        </ModelProvider>
+        </ConfigProvider>
       </NavProvider>
     </main>
   )
 }
 
 function NavFrame() {
-  const port = usePermissionPort()
+  const permissionPort = usePermissionPort()
+  const isPermissionRequest = !!permissionPort.data
   const { view, setSettingsShown, settingsShown } = useNav()
+  const timeout = useRef<NodeJS.Timeout>()
 
   useEffect(() => {
     async function checkConfig() {
+      if (isPermissionRequest) {
+        return
+      }
       const config = await configManager.getDefault()
-      if (configManager.isIncomplete(config)) {
-        // This logic allows us to default the settings page on for first-time users
+      // This logic allows us to default the settings page on for first-time users
+      if (!configManager.isCredentialed(config)) {
         setSettingsShown(true)
       }
     }
-    checkConfig()
-  }, [])
+    // HACK: wait for permission request to be set
+    if (timeout.current) {
+      clearTimeout(timeout.current)
+    }
+    timeout.current = setTimeout(checkConfig, 50)
```

Comment:
So after `isPermissionRequest` mutates, we must wait 50ms before checkConfig can run. 

What happened within that 50ms that'd otherwise fail the operation?

Reply:
For a couple milliseconds, `isPermissionRequest` is false when we're making a permission request because the permission port needs to respond to the call in `usePermissionPort`. We don't want to open the settings page unless this is NOT a permission request AND the default config is not credentialed

Reply:
Perhaps a state value in usePermissionPort can be used to track the status of the request, and the useEffect can track that state to re-render with it as a condition.

Reply:
Good idea, looking into it